Add support for ReadableByteStreamController

youennf opened the pull request:
#### 8734c7e0184da58f57dcb60da2b1de607937299d
<pre>
Add support for ReadableByteStreamController
<a href="https://bugs.webkit.org/show_bug.cgi?id=299426">https://bugs.webkit.org/show_bug.cgi?id=299426</a>
<a href="https://rdar.apple.com/problem/161234313">rdar://problem/161234313</a>

Reviewed by Chris Dumez.

* LayoutTests/imported/w3c/web-platform-tests/streams/idlharness.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/idlharness.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/idlharness.any.sharedworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/idlharness.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/piping/abort.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/piping/abort.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/piping/abort.any.sharedworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/piping/abort.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/bad-buffers-and-views.any.sharedworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/construct-byob-request.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/construct-byob-request.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/construct-byob-request.any.sharedworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/construct-byob-request.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/enqueue-with-detached-buffer.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/enqueue-with-detached-buffer.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/enqueue-with-detached-buffer.any.sharedworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/enqueue-with-detached-buffer.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/general.any.sharedworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/non-transferable-buffers.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/non-transferable-buffers.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/non-transferable-buffers.any.sharedworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/non-transferable-buffers.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/patched-global.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/patched-global.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/patched-global.any.sharedworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/patched-global.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/read-min.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/read-min.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/read-min.any.sharedworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/read-min.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/respond-after-enqueue.any.sharedworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/tee.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/tee.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/tee.any.sharedworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/tee.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/templated.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/templated.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/templated.any.sharedworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/templated.any.worker-expected.txt:
* LayoutTests/streams/readable-stream-byob-reader-expected.txt:
* LayoutTests/streams/readable-stream-byob-reader-worker-expected.txt:
* LayoutTests/streams/readable-stream-byob-request-expected.txt:
* LayoutTests/streams/readable-stream-byob-request-worker-expected.txt:
* Source/WebCore/Modules/streams/ReadableByteStreamController.cpp:
(WebCore::getAlgorithmPromise):
(WebCore::ReadableByteStreamController::ReadableByteStreamController):
(WebCore::ReadableByteStreamController::byobRequestForBindings const):
(WebCore::ReadableByteStreamController::desiredSize const):
(WebCore::ReadableByteStreamController::closeForBindings):
(WebCore::ReadableByteStreamController::enqueueForBindings):
(WebCore::ReadableByteStreamController::errorForBindings):
(WebCore::ReadableByteStreamController::getByobRequest const):
(WebCore::ReadableByteStreamController::getDesiredSize const):
(WebCore::ReadableByteStreamController::start):
(WebCore::ReadableByteStreamController::didStart):
(WebCore::ReadableByteStreamController::close):
(WebCore::transferArrayBuffer):
(WebCore::ReadableByteStreamController::enqueue):
(WebCore::ReadableByteStreamController::processReadRequestsUsingQueue):
(WebCore::ReadableByteStreamController::invalidateByobRequest):
(WebCore::ReadableByteStreamController::processPullIntoDescriptorsUsingQueue):
(WebCore::ReadableByteStreamController::enqueueDetachedPullIntoToQueue):
(WebCore::ReadableByteStreamController::shiftPendingPullInto):
(WebCore::ReadableByteStreamController::enqueueChunkToQueue):
(WebCore::cloneArrayBuffer):
(WebCore::ReadableByteStreamController::enqueueClonedChunkToQueue):
(WebCore::ReadableByteStreamController::callPullIfNeeded):
(WebCore::ReadableByteStreamController::shouldCallPull):
(WebCore::copyDataBlockBytes):
(WebCore::ReadableByteStreamController::fillPullIntoDescriptorFromQueue):
(WebCore::ReadableByteStreamController::fillHeadPullIntoDescriptor):
(WebCore::ReadableByteStreamController::commitPullIntoDescriptor):
(WebCore::ReadableByteStreamController::convertPullIntoDescriptor):
(WebCore::ReadableByteStreamController::error):
(WebCore::ReadableByteStreamController::clearPendingPullIntos):
(WebCore::ReadableByteStreamController::clearAlgorithms):
(WebCore::ReadableByteStreamController::pullInto):
(WebCore::ReadableByteStreamController::runCancelSteps):
(WebCore::ReadableByteStreamController::runPullSteps):
(WebCore::ReadableByteStreamController::releaseSteps):
(WebCore::ReadableByteStreamController::fillReadRequestFromQueue):
(WebCore::ReadableByteStreamController::storeError):
(WebCore::ReadableByteStreamController::storedError const):
(WebCore::ReadableByteStreamController::respond):
(WebCore::ReadableByteStreamController::respondWithNewView):
(WebCore::ReadableByteStreamController::respondInternal):
(WebCore::ReadableByteStreamController::respondInClosedState):
(WebCore::ReadableByteStreamController::respondInReadableState):
(WebCore::ReadableByteStreamController::commitPulllIntoDescriptor):
(WebCore::ReadableByteStreamController::handleQueueDrain):
(WebCore::ReadableByteStreamController::handleSourcePromise):
(WebCore::JSReadableByteStreamController::visitAdditionalChildren):
(WebCore::ReadableByteStreamController::runReleaseSteps): Deleted.
* Source/WebCore/Modules/streams/ReadableByteStreamController.h:
(WebCore::ReadableByteStreamController::underlyingSource):
(WebCore::ReadableByteStreamController::storedErrorObject):
(WebCore::ReadableByteStreamController::pullAlgorithmConcurrently):
(WebCore::ReadableByteStreamController::cancelAlgorithmConcurrently):
(WebCore::ReadableByteStreamController::hasPendingPullIntos const):
* Source/WebCore/Modules/streams/ReadableByteStreamController.idl:
* Source/WebCore/Modules/streams/ReadableStream.cpp:
(WebCore::ReadableStream::createFromJSValues):
(WebCore::ReadableStream::createFromByteUnderlyingSource):
(WebCore::ReadableStream::getReader):
(WebCore::ReadableStream::tee):
(WebCore::ReadableStream::isLocked const):
(WebCore::ReadableStream::cancel):
(WebCore::ReadableStream::pipeTo):
(WebCore::ReadableStream::createReadableByteStream):
(WebCore::ReadableStream::fulfillReadRequest):
(WebCore::ReadableStream::setByobReader):
(WebCore::ReadableStream::fulfillReadIntoRequest):
(WebCore::ReadableStream::setupReadableByteStreamControllerFromUnderlyingSource):
(WebCore::ReadableStream::setupReadableByteStreamController):
(WebCore::ReadableStream::close):
(WebCore::ReadableStream::error):
(WebCore::ReadableStream::getNumReadIntoRequests const):
(WebCore::ReadableStream::addReadIntoRequest):
(WebCore::ReadableStream::pipeThrough):
(WebCore::ReadableStream::storedError const):
(WebCore::JSReadableStream::cancel):
(WebCore::JSReadableStream::visitAdditionalChildren):
* Source/WebCore/Modules/streams/ReadableStream.h:
(WebCore::ReadableStream::controller):
(WebCore::ReadableStream::protectedController):
(WebCore::ReadableStream::pipeTo): Deleted.
* Source/WebCore/Modules/streams/ReadableStream.idl:
* Source/WebCore/Modules/streams/ReadableStreamBYOBReader.cpp:
(WebCore::ReadableStreamBYOBReader::closed):
(WebCore::ReadableStreamBYOBReader::read):
(WebCore::ReadableStreamBYOBReader::setupBYOBReader):
(WebCore::ReadableStreamBYOBReader::genericRelease):
(WebCore::ReadableStreamBYOBReader::onClosedPromiseRejection):
(WebCore::ReadableStreamBYOBReader::closedPromise): Deleted.
* Source/WebCore/Modules/streams/ReadableStreamBYOBReader.h:
* Source/WebCore/Modules/streams/ReadableStreamBYOBReader.idl:
* Source/WebCore/Modules/streams/ReadableStreamDefaultReader.cpp:
(WebCore::ReadableStreamDefaultReader::genericRelease):

Originally-landed-as: e26a2b9644a4. <a href="https://rdar.apple.com/problem/161234313">rdar://problem/161234313</a>
Canonical link: <a href="https://commits.webkit.org/300749@main">https://commits.webkit.org/300749@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a22ae7e7a47b419e4a848a8adc013e9a8617138f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/123756 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/43471 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/34168 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/130534 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/75905 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/1a99fa38-c5c8-4930-8892-53c748c4c940) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/125633 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/44195 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/52065 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/94119 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/62458 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/d42b0f82-0f1b-4436-ae60-7f0c40dc60dc) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/126709 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/35212 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/110713 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/74722 "Found 2 new API test failures: TestWTF:WTF_Condition.OneProducerOneConsumerOneSlotTimeout, WPE/TestWebKitWebContext:/webkit/WebKitWebContext/no-web-process-leak (failure)") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/8f9d1132-2fcb-45ae-adb1-1be1750ac423) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/34170 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/28872 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/74014 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/104936 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/29096 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/133224 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/50708 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/38616 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/102597 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/51083 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/106932 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/102426 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26034 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/47766 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/26018 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/47552 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/50562 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/56326 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/50036 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/53382 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/51710 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/de48b2e9fea3b44a5f7d5b8138b86e32fe6434fc. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51252

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/75fa6f35e21a5bbfc3b3d9cef4ee4b94e6e79f27. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51252

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/60e810b4fa935e314337cfee5c1b05716fd5303b. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51252

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/01bf8feaa73e067a158d56dbf2c2e79e08de90a8. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51252

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/27a23905e272671f76dad763ae849f4ec4001200. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51252

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/6a0216c8057300888e714ced25868342dfad3fad. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51252

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/1ed33e5fb629df86312dfc8ab39e2cf672cb0d94. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51252

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/56a4d2b321b653e7959668f0a9c8f7f7208f5117. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51252

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/0b691fe0b9994f96891dd378c36cb77aa956230c. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51252

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/107c61995ceb50a31a0028463fc904a6bc87168e. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51252

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/ac251352c9b052c2a0201ed3ad05479b22b6482f. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51252

youennf commented on Source/WebCore/Modules/streams/ReadableByteStreamController.cpp:
```diff
@@ -57,107 +111,861 @@ ReadableStream& ReadableByteStreamController::stream()
+}
+
+// https://streams.spec.whatwg.org/#readable-byte-stream-controller-commit-pull-into-descriptor
+void ReadableByteStreamController::commitPulllIntoDescriptor(JSDOMGlobalObject& globalObject, PullIntoDescriptor& pullIntoDescriptor)
```
ok

youennf commented on Source/WebCore/Modules/streams/ReadableByteStreamController.cpp:
```diff
@@ -57,107 +111,861 @@ ReadableStream& ReadableByteStreamController::stream()
+
+    ASSERT(state == ReadableStream::State::Readable);
+    ASSERT(pullIntoDescriptor.readerType != ReaderType::None);
+    bool done = false;
```
done is the name of the chunk member so it is better to keep it like this.

youennf commented on Source/WebCore/Modules/streams/ReadableByteStreamController.h:
```diff
@@ -74,15 +77,101 @@ class ReadableByteStreamController : public CanMakeWeakPtr<ReadableByteStreamCon
+    bool m_pulling { false };
+    mutable RefPtr<ReadableStreamBYOBRequest> m_byobRequest;
+    bool m_closeRequested { false };
+    bool m_started { false };
```
started is the name of the slot in the spec.

youennf commented on Source/WebCore/Modules/streams/ReadableStream.cpp:
```diff
@@ -266,22 +295,38 @@ ReadableStreamBYOBReader* ReadableStream::byobReader()
 
+void ReadableStream::setupReadableByteStreamController(JSDOMGlobalObject& globalObject, ReadableByteStreamController::PullAlgorithm&& pullAlgorithm, ReadableByteStreamController::CancelAlgorithm&& cancelAlgorithm, double highWaterMark)
+{
+    ASSERT(!m_controller);
```
ok

youennf commented on Source/WebCore/Modules/streams/ReadableStreamBYOBReader.cpp:
```diff
@@ -74,8 +78,14 @@ void ReadableStreamBYOBReader::read(JSDOMGlobalObject& globalObject, JSC::ArrayB
+            return promise->reject(Exception { ExceptionCode::RangeError, "view's buffer is not long enough"_s });
+    } else {
+        if (options.min > view.byteLength())
+            return promise->reject(Exception { ExceptionCode::RangeError, "view's buffer is not long enough"_s });
```
ok

youennf commented on Source/WebCore/Modules/streams/ReadableStreamBYOBReader.cpp:
```diff
@@ -74,8 +78,14 @@ void ReadableStreamBYOBReader::read(JSDOMGlobalObject& globalObject, JSC::ArrayB
+    auto viewType = view.getType();
+    if (viewType != JSC::TypedArrayType::TypeDataView) {
+        if (options.min > view.byteLength() / JSC::elementSize(viewType))
+            return promise->reject(Exception { ExceptionCode::RangeError, "view's buffer is not long enough"_s });
```
ok

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/5eba8f244de0b860dda09b6a13d7e6875951bc7a. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51252

youennf committed:
Add support for ReadableByteStreamController
https://bugs.webkit.org/show_bug.cgi?id=299426
rdar://problem/161234313

Reviewed by Chris Dumez.

* LayoutTests/imported/w3c/web-platform-tests/streams/idlharness.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/idlharness.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/idlharness.any.sharedworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/idlharness.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/piping/abort.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/piping/abort.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/piping/abort.any.sharedworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/piping/abort.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/bad-buffers-and-views.any.sharedworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/construct-byob-request.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/construct-byob-request.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/construct-byob-request.any.sharedworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/construct-byob-request.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/enqueue-with-detached-buffer.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/enqueue-with-detached-buffer.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/enqueue-with-detached-buffer.any.sharedworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/enqueue-with-detached-buffer.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/general.any.sharedworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/non-transferable-buffers.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/non-transferable-buffers.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/non-transferable-buffers.any.sharedworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/non-transferable-buffers.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/patched-global.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/patched-global.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/patched-global.any.sharedworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/patched-global.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/read-min.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/read-min.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/read-min.any.sharedworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/read-min.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/respond-after-enqueue.any.sharedworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/tee.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/tee.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/tee.any.sharedworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/tee.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/templated.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/templated.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/templated.any.sharedworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/readable-byte-streams/templated.any.worker-expected.txt:
* LayoutTests/streams/readable-stream-byob-reader-expected.txt:
* LayoutTests/streams/readable-stream-byob-reader-worker-expected.txt:
* LayoutTests/streams/readable-stream-byob-request-expected.txt:
* LayoutTests/streams/readable-stream-byob-request-worker-expected.txt:
* Source/WebCore/Modules/streams/ReadableByteStreamController.cpp:
(WebCore::getAlgorithmPromise):
(WebCore::ReadableByteStreamController::ReadableByteStreamController):
(WebCore::ReadableByteStreamController::byobRequestForBindings const):
(WebCore::ReadableByteStreamController::desiredSize const):
(WebCore::ReadableByteStreamController::closeForBindings):
(WebCore::ReadableByteStreamController::enqueueForBindings):
(WebCore::ReadableByteStreamController::errorForBindings):
(WebCore::ReadableByteStreamController::getByobRequest const):
(WebCore::ReadableByteStreamController::getDesiredSize const):
(WebCore::ReadableByteStreamController::start):
(WebCore::ReadableByteStreamController::didStart):
(WebCore::ReadableByteStreamController::close):
(WebCore::transferArrayBuffer):
(WebCore::ReadableByteStreamController::enqueue):
(WebCore::ReadableByteStreamController::processReadRequestsUsingQueue):
(WebCore::ReadableByteStreamController::invalidateByobRequest):
(WebCore::ReadableByteStreamController::processPullIntoDescriptorsUsingQueue):
(WebCore::ReadableByteStreamController::enqueueDetachedPullIntoToQueue):
(WebCore::ReadableByteStreamController::shiftPendingPullInto):
(WebCore::ReadableByteStreamController::enqueueChunkToQueue):
(WebCore::cloneArrayBuffer):
(WebCore::ReadableByteStreamController::enqueueClonedChunkToQueue):
(WebCore::ReadableByteStreamController::callPullIfNeeded):
(WebCore::ReadableByteStreamController::shouldCallPull):
(WebCore::copyDataBlockBytes):
(WebCore::ReadableByteStreamController::fillPullIntoDescriptorFromQueue):
(WebCore::ReadableByteStreamController::fillHeadPullIntoDescriptor):
(WebCore::ReadableByteStreamController::commitPullIntoDescriptor):
(WebCore::ReadableByteStreamController::convertPullIntoDescriptor):
(WebCore::ReadableByteStreamController::error):
(WebCore::ReadableByteStreamController::clearPendingPullIntos):
(WebCore::ReadableByteStreamController::clearAlgorithms):
(WebCore::ReadableByteStreamController::pullInto):
(WebCore::ReadableByteStreamController::runCancelSteps):
(WebCore::ReadableByteStreamController::runPullSteps):
(WebCore::ReadableByteStreamController::releaseSteps):
(WebCore::ReadableByteStreamController::fillReadRequestFromQueue):
(WebCore::ReadableByteStreamController::storeError):
(WebCore::ReadableByteStreamController::storedError const):
(WebCore::ReadableByteStreamController::respond):
(WebCore::ReadableByteStreamController::respondWithNewView):
(WebCore::ReadableByteStreamController::respondInternal):
(WebCore::ReadableByteStreamController::respondInClosedState):
(WebCore::ReadableByteStreamController::respondInReadableState):
(WebCore::ReadableByteStreamController::commitPulllIntoDescriptor):
(WebCore::ReadableByteStreamController::handleQueueDrain):
(WebCore::ReadableByteStreamController::handleSourcePromise):
(WebCore::JSReadableByteStreamController::visitAdditionalChildren):
(WebCore::ReadableByteStreamController::runReleaseSteps): Deleted.
* Source/WebCore/Modules/streams/ReadableByteStreamController.h:
(WebCore::ReadableByteStreamController::underlyingSource):
(WebCore::ReadableByteStreamController::storedErrorObject):
(WebCore::ReadableByteStreamController::pullAlgorithmConcurrently):
(WebCore::ReadableByteStreamController::cancelAlgorithmConcurrently):
(WebCore::ReadableByteStreamController::hasPendingPullIntos const):
* Source/WebCore/Modules/streams/ReadableByteStreamController.idl:
* Source/WebCore/Modules/streams/ReadableStream.cpp:
(WebCore::ReadableStream::createFromJSValues):
(WebCore::ReadableStream::createFromByteUnderlyingSource):
(WebCore::ReadableStream::getReader):
(WebCore::ReadableStream::tee):
(WebCore::ReadableStream::isLocked const):
(WebCore::ReadableStream::cancel):
(WebCore::ReadableStream::pipeTo):
(WebCore::ReadableStream::createReadableByteStream):
(WebCore::ReadableStream::fulfillReadRequest):
(WebCore::ReadableStream::setByobReader):
(WebCore::ReadableStream::fulfillReadIntoRequest):
(WebCore::ReadableStream::setupReadableByteStreamControllerFromUnderlyingSource):
(WebCore::ReadableStream::setupReadableByteStreamController):
(WebCore::ReadableStream::close):
(WebCore::ReadableStream::error):
(WebCore::ReadableStream::getNumReadIntoRequests const):
(WebCore::ReadableStream::addReadIntoRequest):
(WebCore::ReadableStream::pipeThrough):
(WebCore::ReadableStream::storedError const):
(WebCore::JSReadableStream::cancel):
(WebCore::JSReadableStream::visitAdditionalChildren):
* Source/WebCore/Modules/streams/ReadableStream.h:
(WebCore::ReadableStream::controller):
(WebCore::ReadableStream::protectedController):
(WebCore::ReadableStream::pipeTo): Deleted.
* Source/WebCore/Modules/streams/ReadableStream.idl:
* Source/WebCore/Modules/streams/ReadableStreamBYOBReader.cpp:
(WebCore::ReadableStreamBYOBReader::closed):
(WebCore::ReadableStreamBYOBReader::read):
(WebCore::ReadableStreamBYOBReader::setupBYOBReader):
(WebCore::ReadableStreamBYOBReader::genericRelease):
(WebCore::ReadableStreamBYOBReader::onClosedPromiseRejection):
(WebCore::ReadableStreamBYOBReader::closedPromise): Deleted.
* Source/WebCore/Modules/streams/ReadableStreamBYOBReader.h:
* Source/WebCore/Modules/streams/ReadableStreamBYOBReader.idl:
* Source/WebCore/Modules/streams/ReadableStreamDefaultReader.cpp:
(WebCore::ReadableStreamDefaultReader::genericRelease):

Originally-landed-as: e26a2b9644a4. rdar://problem/161234313
Canonical link: https://commits.webkit.org/300749@main
